ioctl/cmd/bc/bcbucketlist.go

```diff
@@ -7,47 +7,60 @@
 package bc
 
 import (
-	"context"
 	"fmt"
+	"strconv"
 	"strings"
 
-	"github.com/golang/protobuf/proto"
-	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
-	"github.com/spf13/cobra"
-	"google.golang.org/grpc/status"
-
 	"github.com/iotexproject/iotex-proto/golang/iotexapi"
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/spf13/cobra"
 
 	"github.com/iotexproject/iotex-core/ioctl/config"
 	"github.com/iotexproject/iotex-core/ioctl/output"
 	"github.com/iotexproject/iotex-core/ioctl/util"
 )
 
+const (
+	bucketlistMethodByVoter     = "voter"
+	bucketlistMethodByCandidate = "cand"
+)
+
+var (
+	validMethods = []string{
+		bucketlistMethodByVoter,
+		bucketlistMethodByCandidate,
+	}
+)
+
 // Multi-language support
 var (
 	bcBucketListCmdShorts = map[config.Language]string{
-		config.English: "Get bucket list for given address on IoTeX blockchain",
-		config.Chinese: "在IoTeX区块链上读取账户地址的投票列表",
+		config.English: "Get bucket list with method and arg(s) on IoTeX blockchain",
+		config.Chinese: "根据方法和参数在IoTeX区块链上读取投票列表",
+	}
+	bcBucketListCmdUses = map[config.Language]string{
+		config.English: "bucketlist <method> [arguments]",
+		config.Chinese: "bucketlist <方法> [参数]",
 	}
-	bcBucketListUses = map[config.Language]string{
-		config.English: "bucketlist [ALIAS|ADDRESS]",
-		config.Chinese: "bucketlist [别名|地址]",
+	bcBucketListCmdLongs = map[config.Language]string{
+		config.English: "Read bucket list\nValid methods: [" +
+			strings.Join(validMethods, ", ") + "]",
+		config.Chinese: "根据方法和参数在IoTeX区块链上读取投票列表\n可用方法有：" +
+			strings.Join(validMethods, "，"),
 	}
 )
 
 // bcBucketListCmd represents the bc bucketlist command
 var bcBucketListCmd = &cobra.Command{
-	Use:   config.TranslateInLang(bcBucketListUses, config.UILanguage),
+	Use:   config.TranslateInLang(bcBucketListCmdUses, config.UILanguage),
 	Short: config.TranslateInLang(bcBucketListCmdShorts, config.UILanguage),
-	Args:  cobra.RangeArgs(0, 1),
+	Long:  config.TranslateInLang(bcBucketListCmdLongs, config.UILanguage),
+	Args:  cobra.MinimumNArgs(2),
+	Example: `ioctl bc bucketlist voter [VOTER_ADDRESS] [OFFSET] [LIMIT]
+ioctl bc bucketlist cand [CANDIDATE_NAME] [OFFSET] [LIMIT]`,
 	RunE: func(cmd *cobra.Command, args []string) error {
 		cmd.SilenceUsage = true
-		arg := ""
-		if len(args) == 1 {
-			arg = args[0]
-		}
-		err := getBucketList(arg)
+		err := getBucketList(args[0], args[1:]...)
```

Comment:
since it's minArgs(2) now, args[0] = method, args[1] = addr must exist
so can do getBucketList(args[0], args[1], args[2:]...)